server/src/main/java/org/opensearch/index/mapper/DateFieldMapper.java

```diff
@@ -91,7 +91,14 @@ public final class DateFieldMapper extends ParametrizedFieldMapper {
 
     public static final String CONTENT_TYPE = "date";
     public static final String DATE_NANOS_CONTENT_TYPE = "date_nanos";
-    public static final DateFormatter DEFAULT_DATE_TIME_FORMATTER = DateFormatter.forPattern("strict_date_optional_time||epoch_millis");
+    @Deprecated
+    public static final DateFormatter LEGACY_DEFAULT_DATE_TIME_FORMATTER = DateFormatter.forPattern(
+        "strict_date_optional_time||epoch_millis"
+    );
+    public static final DateFormatter DEFAULT_DATE_TIME_FORMATTER = DateFormatter.forPattern(
```

Comment:
@CaptainDredge please correct me if I am wrong, but it seems like we are bundling 2 independent changes here: 

1) the parser optimization
2) widening default formatter patterns

I believe the best option would be to split these changes in 2 changesets, since the feature flag we've introduced only controls the caching behavior.

Reply:
@reta yes, we're bundling both the changes but the parser optimisation is only applicable on the default formatter and we're not changing the order of formatters in case of user specified format, the overall perf gain are visible only in the default case where the gains are mainly coming from the fact that a stricter formatter is being applied before the optional one. Therefore, I bundled them together as both together provides the perf gains. What are some issues you see in bundling these changes together? I personally don't see any major benefit in separating this in 2 PRs

Reply:
Thanks for the context @CaptainDredge , the main issue I see is that we have introduced a feature flag to guard  this change against but it only partially reversible. I am fine with keeping it in there but I think we should:
 - revert to LEGACY_DEFAULT_DATE_TIME_FORMATTER if feature flag is not set
 - add more tests for `rest-api-spec` to test different date format combinations

Reply:
Makes sense, let me make changes to revert back to legacy formatter if the feature flag is not set

Reply:
I have no more comments, just looking through the changes